utils/scripts/install-dev.sh

```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+
+# This script sets up a new development node with all dependencies needed to
+# build DAOS from source.
+
+# Requirements:
+# - Run as a non-root user with sudo access.
+
+if [ "$(whoami)" == "root" ]; then
+	echo "Run this script as the non-root user you will use to build DAOS"
+	exit 1
+fi
+
+distro=$1
+case "$distro" in
+	centos7 | el8 | leap15 | ubuntu)
+		echo "Setting up node for $distro...";;
+
+	*)
+		echo "Invalid distro option: \"$distro\""
+		echo "syntax: $0 <el8|leap15|ubuntu|centos7>"
+		exit 1
+esac
+
+# Install dependencies from the OS repos.
+SCRIPT_DIR="$(dirname "$(realpath -e "${BASH_SOURCE[0]}")")"
+sudo "$SCRIPT_DIR/install-$distro.sh" && \
+# Install user dependencies
+python3 -m pip install -r "$SCRIPT_DIR/../../requirements.txt"
```

Comment:
This isn't the same as I described at all, and likely would not work.  There are three ways of running pip, and they are:
1. Run it as root and install into /usr/local
2. Run it as a user and install into ~/.local
3. Setup a venv in advance and run it as user to install into the venv.
It's not clear what this script is doing, if the sudo applies from L27 applies here it'll do 1. which would work but would potentially mess up installation of future OS packages, if the sudo doesn't apply then it'll do 2. which will work for python packages but unless you've added ~/.local/python/3.8/bin (or similar) to your PATH then it will only help with python libraries, not python commands like meson.

To install into a venv you'd need to do the following:
```suggestion
python3 -m venv "$SCRIPT_DIR/venv"
source "$SCRIPT_DIR/venv/bin/activate"
python3 -m pip install -r "$SCRIPT_DIR/../../requirements.txt"
```
but then you'd have to call the `source` command from your shell for each new shell you want to be able to access the packages.

Reply:
My intention was item 2 -- seemed the simplest to me. Is there a good reason to use venv for build dependencies as a DAOS dev, rather than installing them generally at the user level? It seems like both require an update to the user's environment (either updating the PATH or sourcing the venv).

If we want to standardize on the venv, I don't think we'd want to put the virtualenv in the DAOS tree (where $SCRIPT_DIR is), but instead somewhere in the user's home (`$HOME/daosvenv`?).